library/core/src/mem/mod.rs

```diff
@@ -659,14 +659,9 @@ pub unsafe fn zeroed<T>() -> T {
 /// correctly: it has the same effect as [`MaybeUninit::uninit().assume_init()`][uninit].
 /// As the [`assume_init` documentation][assume_init] explains,
 /// [the Rust compiler assumes][inv] that values are properly initialized.
-/// As a consequence, calling e.g. `mem::uninitialized::<bool>()` causes immediate
-/// undefined behavior for returning a `bool` that is not definitely either `true`
-/// or `false`. Worse, truly uninitialized memory like what gets returned here
-/// is special in that the compiler knows that it does not have a fixed value.
-/// This makes it undefined behavior to have uninitialized data in a variable even
-/// if that variable has an integer type.
```

Comment:
I think it would be nice to keep this part of the explanation. Unless we have a better explanation of _why_ this is UB. I think a lot of programmers from a C/C++ background will find this very unintuitive and I think it would be much more welcoming to let them know how much consideration this was given.

Reply:
I re-added some of the justification here. I'm leaving off the mem::uninit<bool> example since that implies uninit u8's are fine, which is the opposite of what we're saying here.

Though as for concrete justification as to *why*....

Uhhhh... It makes the model simpler? It does mean we don't need to answer questions about "what happens to uninit bytes when you load/store them". And it lets us emit noundef more, though I don't know how much llvm cares about that at the moment.

Reply:
> It does mean we don't need to answer questions about "what happens to uninit bytes when you load/store them". 

We still need to say that since they can occur inside unions.